src/utils/Router.js

```diff
@@ -0,0 +1,36 @@
+import { Router } from 'express';
+import permissionsValidator from '../validators';
+
+const methods = ['post', 'get', 'put', 'delete'];
+
+class ExtendRouter {
+  constructor() {
+    let router = Router();
+
+    router = this.setMethods(router);
+
+    return router;
+  }
+
+  setMethods(router) {
+    const modifyRouter = router;
+    methods.forEach((method) => {
+      modifyRouter[method] = this.setMethod(method);
+    });
+    return modifyRouter;
```

Comment:
There's no point in doing this, you will still modify original router object (JS reference objects)
I made this example to demo it -> https://jsbin.com/bewumulori/edit?html,js,console

In this case it's OK to mutate an argument, so you call this.setMethods(new Router()) in the constructor;

returning value from a consturctor in this case simply creates a singleton -> read about it [here](https://web.archive.org/web/20100216124827/http://www.gibdon.com/2008/08/javascript-constructor-return-value.html)